dashboards-observability/public/components/visualizations/charts/dashboard_list/dashboard_list_type.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { getPlotlySharedConfigs, getPlotlyCategory } from '../shared/shared_configs';
+import { LensIconChartLine } from '../../assets/chart_line';
+import { PLOTLY_COLOR } from '../../../../../common/constants/shared';
+import { DashboardList } from './dashboard_list';
+import { VizDataPanel } from '../../../event_analytics/explorer/visualizations/config_panel/config_panes/default_vis_editor';
+import { ConfigText } from '../../../event_analytics/explorer/visualizations/config_panel/config_panes/config_controls';
+
+const sharedConfigs = getPlotlySharedConfigs();
+const VIS_CATEGORY = getPlotlyCategory();
+
+export const createDashboardListTypeDefinition = (params: any = {}) => ({
+  name: 'dashboard_list',
+  type: 'dashboard_list',
+  id: 'dashboard_list',
+  label: 'Dashboard list',
+  fulllabel: 'Dashboard list',
+  icontype: 'visTable',
+  category: VIS_CATEGORY.BASICS,
+  selection: {
+    dataLoss: 'nothing',
+  },
+  icon: LensIconChartLine,
+  categoryaxis: 'xaxis',
+  seriesaxis: 'yaxis',
+  editorconfig: {
+    panelTabs: [
+      {
+        id: 'data-panel',
+        name: 'Style',
+        mapTo: 'dataConfig',
+        editor: VizDataPanel,
+        sections: [
+          {
+            id: 'text_editor',
+            name: 'Text',
+            editor: ConfigText,
+            mapTo: 'text',
+            schemas: [],
+          },
+        ],
+      },
+    ],
+  },
+  visconfig: {
+    layout: {
+      ...sharedConfigs.layout,
+      ...{
+        colorway: PLOTLY_COLOR,
+        plot_bgcolor: 'rgba(0, 0, 0, 0)',
+        paper_bgcolor: 'rgba(0, 0, 0, 0)',
+        xaxis: {
+          fixedrange: true,
+          showgrid: false,
+          visible: true,
+        },
+        yaxis: {
+          fixedrange: true,
+          showgrid: false,
+          visible: true,
+        },
+      },
+    },
+    config: {
+      ...sharedConfigs.config,
+      ...{
+        barmode: 'line',
+        xaxis: {
+          automargin: true,
+        },
+        yaxis: {
+          automargin: true,
+        },
```

Comment:
isn't it same as the below? Why to create an object to spread that out? Is this new syntax helpful now a days, and I am missing something? I see many of you writing code of this type now a days.
```JS
...sharedConfigs.config,
barmode: 'line',
xaxis: {
  automargin: true,
},
yaxis: {
  automargin: true,
},
```

Reply:
Updated code